_posts/05-06-01-Internationalization-and-Localization.md

```diff
@@ -0,0 +1,348 @@
+    - missing translations would display meaningless keys on screen (`TOP_MENU_WELCOME` instead of `Hello there, User!`
+    on the said untranslated French page). That's good as would force translation to be complete before publishing -
+    but bad as translation issues would be really awful in the interface.
+
+The [Gettext manual][manual] favors the first approach, as in general it's easier for translators and users in
+case of trouble. That's how we will be working here as well.
+
+### Everyday usage
+In a common application, you would use some Gettext functions while writing static text in your pages. Those sentences
+would then appear in `.po` files, get translated, compiled into `.mo` files and then, used by Gettext when rendering
+the actual interface. Given that, let's tie together what we have discussed so far in a a step-by-step example:
+
+#### 1. A sample template file, including some different gettext calls
+{% highlight php %}
+<?php include 'i18n_setup.php' ?>
+<div id="header">
+    <h1><?=sprintf(gettext('Welcome, %s!'), $name)?></h1>
+    <!-- code indented this way only for legibility here -->
+    <?php if ($unread): ?>
+        <h2><?=sprintf(
+            ngettext('Only one unread message',
+                     '%d unread messages',
+                     $unread),
+            $unread)?>
+        </h2>
+    <? endif ?>
+</div>
+
+<h1><?=gettext('Introduction')?></h1>
+<p><?=gettext('We\'re now translating some strings')?></p>
+{% endhighlight %}
+
+- [`gettext()`][func] simply translates a `msgid` into it's corresponding `msgstr` for a given language. There's also
+the shorthand function `_()` that works the same way;
+- [`ngettext()`][n_func] does the same but with plural rules;
+- there's also [`dgettext()`][d_func] and [`dngettext()`][dn_func], that allows you to override the domain for a single
+call. More on domain configuration in the next example.
+
+#### 2. A sample setup file (`i18n_setup.php` as used above), selecting the correct locale and configuring Gettext
+{% highlight php %}
+<?php
+/**
+ * Verifies if the given $locale is supported in the project
+ * @param string $locale
+ * @return bool
+ */
+function valid($locale) {
+   return in_array($locale, ['en_US', 'en', 'pt_BR', 'pt', 'es_ES', 'es');
+}
+
+//setting the source/default locale, for informational purposes
+$lang = 'en_US';
+
+if (isset($_GET['lang']) && valid($_GET['lang'])) {
+    // the locale can be changed through the query-string
+    $lang = $_GET['lang'];    //you should sanitize this!
+    setcookie('lang', $lang); //it's stored in a cookie so it can be reused
+} elseif (isset($_COOKIE['lang']) && valid($_COOKIE['lang'])) {
+    // if the cookie is present instead, let's just keep it
+    $lang = $_COOKIE['lang']; //you should sanitize this!
+} elseif (isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
+    // default resort: look for the languages the browser says the user accepts
```

Comment:
>    // default resort: look for the languages the browser says the user accepts

I'd recommend removing the word "resort", as it's not needed and in this context is used awkwardly.